extensions/libarchive/BinFiles.h

```diff
@@ -87,8 +87,10 @@ class Bin {
       }
     }
 
-    if ((queued_data_size_ + flow->getSize()) > maxSize_ || (queue_.size() + 1) > maxEntries_)
+    if ((queued_data_size_ + flow->getSize()) > maxSize_ || (queue_.size() + 1) > maxEntries_) {
+      closed_ = true;
```

Comment:
Do you think it would make sense to penalize flow files that are larger than the max group size? Currently if there is such a flow file, it will keep flushing out all bins regardless of how "full" they are.

Reply:
incoming flow files that are larger than max group size are immediately assigned their own bin, and flushed by themselves, so they won't be a problem, flow files that are large but not max size large, could cause the flush of a single bin, as we only try to insert into a single bin (the last in the group's queue) before creating a new bin for it